Linked images increase loserboard

natep-tech opened the pull request:
Adds a Guild setting for letting linked images (image URLs that get embedded) increase the user's Loserboard score.

natep-tech committed:
Add math operations to the /set command, so board values can be modified without explicitly setting the final value.

natep-tech committed:
Allow linked images to increase the Loserboard count (this will be controlled with a per-guild config setting).

natep-tech committed:
Allow linked images to increase the Loserboard count (this will be controlled with a per-guild config setting).

natep-tech committed:
Resolve syntax issues

natep-tech committed:
Merge branch 'LinkedImagesIncreaseLoserboard' of https://github.com/natep-tech/Raiha into LinkedImagesIncreaseLoserboard

9vult commented on src/handlers/messageHandler.ts:
```diff
@@ -10,6 +10,7 @@ import { remindUser } from "../actions/remindUser.action";
 import { informNewAutoModeOptOut } from "../actions/informNewAutoModeOptOut";
 import { urlCheck } from "../actions/urlCheck.action";
 import { urlCheckWarning } from "../actions/urlCheckWarning.action";
+import { urlCheckLoserboard } from "src/actions/urlCheckLoserboard.action";
```
Relative path pls :)
(won't compile with absolute paths. probably something broken in the tsconfig but oh well)

natep-tech committed:
Fix problems with variables
- newRef -> originalValue and is not modified.
- Refactor if/else to change the DB and not self-reference itself.

9vult commented on src/handlers/messageHandler.ts:
```diff
@@ -42,7 +43,10 @@ async function botCallBranch(msg: Message<true>) {
-  if (urlCheck(msg)) urlCheckWarning(msg);
+  if (urlCheck(msg)) {
+    urlCheckWarning(msg);
+    urlCheckLoserboard(msg.guild!.id, msg.author.id);
```
This will work as-is, but for the sake of consistency with the rest of the functions, I would go ahead and pass `msg` through as a single parameter, and pick out the parts you need on the other side.

9vult commented on src/actions/urlCheckLoserboard.action.ts:
```diff
@@ -0,0 +1,10 @@
+import { ServerValue } from "firebase-admin/database";
+import { db, CLIENT, leaderboards } from "../raiha";
+
+export async function urlCheckLoserboard(guildId: string, msgAuthor: Message.author.id<true>) {
```
See the other comment where this function is called.
In that case, the parameter will be of type `Message<true>`, and you can pull out the ids here.

9vult commented on src/listeners/interactionCreate.ts:
```diff
@@ -64,14 +65,24 @@ export default async function (interaction: any) {
       const embed = new EmbedBuilder()
         .setTitle(`Leaderboard Override`)
-        .setDescription(`Set <@${specifiedUser!.id}>'s **${specifiedBoard!}** value to \`${specifiedValue!}\`.`)
+        .setDescription(`Set <@${specifiedUser!.id}>'s **${specifiedBoard!}** value from \`${await ref.val()!}\` to \`${originalValue!}\`.`)
```
other way around lol (from original to new)

9vult reviewed (changes requested):
A couple small things!

natep-tech committed:
Fix issues

9vult commented on src/listeners/interactionCreate.ts:
```diff
@@ -64,14 +65,24 @@ export default async function (interaction: any) {
 
       const ref = db.ref(`/Leaderboard/${specifiedBoard!}/${interaction.guildId}`).child(specifiedUser.id);
-      ref.set(specifiedValue);
+      const originalValue = await ref.val(); //Used to save new value for response message, will be updated later to reflect the arithmetic operation
```
```suggestion
      const originalValue = (await ref.get()).val();
```

9vult commented on src/listeners/interactionCreate.ts:
```diff
@@ -64,14 +65,24 @@ export default async function (interaction: any) {
       const embed = new EmbedBuilder()
         .setTitle(`Leaderboard Override`)
-        .setDescription(`Set <@${specifiedUser!.id}>'s **${specifiedBoard!}** value to \`${specifiedValue!}\`.`)
+        .setDescription(`Set <@${specifiedUser!.id}>'s **${specifiedBoard!}** value from \`${originalValue!}\` to \`${await ref.val()!}\`.`)
```
```suggestion
        .setDescription(`Set <@${specifiedUser!.id}>'s **${specifiedBoard!}** value from \`${originalValue!}\` to \`${(await ref.get()).val()!}\`.`)

```

9vult commented on src/listeners/ready.ts:
```diff
@@ -73,6 +73,15 @@ export default function() {
+    .addChoices(
+      { name: 'Add value to existing', value: 'Add' },
+      { name: 'Subtract value from existing', value: 'Subtract' },
+      { name: 'Replace existing with value', value: 'Absolute' }
```
```suggestion
      { name: 'Add', value: 'Add' },
      { name: 'Subtract', value: 'Subtract' },
      { name: 'Replace', value: 'Absolute' }
```

9vult reviewed (changes requested):
Just this and we good!

natep-tech committed:
Update src/listeners/interactionCreate.ts

Co-authored-by: 9vult <48932637+9vult@users.noreply.github.com>

natep-tech committed:
Update src/listeners/ready.ts

Simplify command options

Co-authored-by: 9vult <48932637+9vult@users.noreply.github.com>

natep-tech committed:
Update src/listeners/interactionCreate.ts

Co-authored-by: 9vult <48932637+9vult@users.noreply.github.com>
